lib/ansible/modules/get_url.py

```diff
@@ -512,7 +512,7 @@ def main():
 
             # Look through each line in the checksum file for a hash corresponding to
             # the filename in the url, returning the first hash that is found.
-            for cksum in (s for (s, f) in checksum_map if f.strip('./') == filename):
+            for cksum in (s for (s, f) in checksum_map if f.strip('./*') == filename):
```

Comment:
one issue, '*' can be a legal characther at the start/end of a file name

Reply:
Yeah, I think we need to improve the logic, to not just strip. But instead normalize the filename by checking if `f[0]` is in `./*` and then only stripping the first char.